extension/src/debug.ts

```diff
@@ -13,62 +13,77 @@ let strNoADPerr: string = AutoLispExt.localize("autolispext.debug.nodap", "doesn
 let strNoACADerr: string = AutoLispExt.localize("autolispext.debug.noacad", "doesn’t exist. Verify and correct the folder path to the product executable.");
 let acadPid2Attach = -1;
 
-let attachCfgName = 'AutoLISP Debug: Attach';
-let attachCfgType = 'attachlisp';
-let attachCfgRequest = 'attach';
+const attachCfgName = 'AutoLISP Debug: Attach';
+const attachCfgType = 'attachlisp';
+const launchCfgName = 'AutoLISP Debug: Attach';
+const launchCfgType = 'launchlisp';
+const attachCfgRequest = 'attach';
 
 export function setDefaultAcadPid(pid: number) {
     acadPid2Attach = pid;
 }
 function need2AddDefaultConfig(config: vscode.DebugConfiguration): Boolean {
-    if (config.type) return false;
-    if (config.request) return false;
-    if (config.name) return false;
+  if (config.type) return false;
+  if (config.request) return false;
+  if (config.name) return false;
 
-    return true;
+  return true;
 }
 
 const LAUNCH_PROC:string = 'debug.LaunchProgram';
 const LAUNCH_PARM:string = 'debug.LaunchParameters';
 const ATTACH_PROC:string = 'debug.AttachProcess';
 
-export function registerLispDebugProviders(context: vscode.ExtensionContext) {
-    //-----------------------------------------------------------
-    //4. debug adapter
-    //-----------------------------------------------------------
-    //this command is used for launch debug calculating DebugAdapter path and arguments
-    context.subscriptions.push(vscode.commands.registerCommand("extension.lispLaunchAdapterExecutableCommand", async () => {
-
-        let lispadapterpath = ProcessPathCache.globalLispAdapterPath;
-        let productStartCommand = ProcessPathCache.globalProductPath;
-        let productStartParameter = ProcessPathCache.globalParameter;
-        return {
-            command: lispadapterpath,
-            args: ["--", productStartCommand, productStartParameter]
-        };
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand("extension.lispAttachAdapterExecutableCommand", async () => {
-        let lispadapterpath = ProcessPathCache.globalLispAdapterPath;
-        return {
-            command: lispadapterpath,
-            args: []
-        };
-    }));
+class LaunchDebugAdapterExecutableFactory
+  implements vscode.DebugAdapterDescriptorFactory {
+  createDebugAdapterDescriptor(
+    _session: vscode.DebugSession,
+    executable: vscode.DebugAdapterExecutable | undefined
+  ): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+    let lispadapterpath = ProcessPathCache.globalLispAdapterPath;
```

Comment:
It looks like a big change, I suggest to request for formal testing before it's merged.

Reply:
Yes, this is a bug changes, but it only affect to start the debugger. I have installed the AutoCAD 2021 and tested the "attach" and "launch" debugging modes